sldns/wire2str.c

```diff
@@ -1139,11 +1141,34 @@ static int sldns_wire2str_svcparam_ech2str(char** s,
 	(*s) += size;
 	(*slen) -= size;
 
-	w += sldns_str_print(s, slen, "\"");	
+	w += sldns_str_print(s, slen, "\"");
 
 	return w + size;
 }
 
+static int sldns_wire2str_svcparam_dohpath2str(char** s,
+	size_t* slen, uint16_t data_len, uint8_t* data)
+{
+	int w = 0;
+	uint16_t i;
+
+	assert(data_len > 0); /* Guaranteed by sldns_wire2str_svcparam_scan */
+
+	w += sldns_str_print(s, slen, "=\"");
+
+	/* RC6570#section-2.1 specifies that the '\' (and other non-letter
+	 * characters in the URI) are "intended to be copied literally" */
+	for (i = 0; i < data_len; i++) {
+		// @TODO do a check like isprint()?
```

Comment:
But the parse routine uses sldns_parse_char for the values, and that unescapes every character. Also I think it is prudent to do an isprint check, so that the output is printable characters.